timelock-server/src/test/java/com/palantir/atlasdb/timelock/lock/HeldLocksCollectionTest.java

```diff
@@ -111,6 +128,106 @@ public void unlockReturnsSubsetOfUnlockedLocks() {
         assertThat(actual).isEqualTo(expected);
     }
 
+    @Test
+    public void lockLeasesAreValidUntilExpiry() {
+        AsyncResult<HeldLocks> result = new AsyncResult<>();
+        AsyncResult<Leased<LockToken>> asyncResult =
+                heldLocksCollection.getExistingOrAcquire(REQUEST_ID, () -> result);
+        result.complete(heldLocksForId(REQUEST_ID));
+
+        Lease lease = asyncResult.get().lease();
+        assertThat(lease.isValid(leaderClock.time())).isTrue();
+
+        setTimeStatic(clock.get().plus(LockLeaseContract.CLIENT_LEASE_TIMEOUT.minus(Duration.ofNanos(1))));
+        assertThat(lease.isValid(leaderClock.time())).isTrue();
+    }
+
+    @Test
+    public void lockLeasesAreInvalidAfterExpiry() {
+        AsyncResult<HeldLocks> result = new AsyncResult<>();
+        AsyncResult<Leased<LockToken>> asyncResult =
+                heldLocksCollection.getExistingOrAcquire(REQUEST_ID, () -> result);
+        result.complete(heldLocksForId(REQUEST_ID));
+
+        Lease lease = asyncResult.get().lease();
+
+        setTimeStatic(clock.get().plus(LockLeaseContract.CLIENT_LEASE_TIMEOUT));
+        assertThat(lease.isValid(leaderClock.time())).isFalse();
+    }
+
+    @Test
+    public void lockLeasesShouldBeInvalidatedBeforeLocksAreReaped() {
+        AsyncResult<HeldLocks> result = new AsyncResult<>();
+        AsyncResult<Leased<LockToken>> asyncResult =
+                heldLocksCollection.getExistingOrAcquire(REQUEST_ID, () -> result);
+        result.complete(heldLocksForId(REQUEST_ID));
+
+        Lease lease = asyncResult.get().lease();
+
+        setTimeStatic(clock.get().plus(LockLeaseContract.CLIENT_LEASE_TIMEOUT));
+        assertThat(lease.isValid(leaderClock.time())).isFalse();
+        assertLocked(REQUEST_ID);
+    }
+
+    @Test
+    public void locksShouldBeReapedAndLeaseShouldBeInvalidatedAfterReapPeriod() {
+        AsyncResult<HeldLocks> result = new AsyncResult<>();
+        AsyncResult<Leased<LockToken>> asyncResult =
+                heldLocksCollection.getExistingOrAcquire(REQUEST_ID, () -> result);
+        result.complete(heldLocksForId(REQUEST_ID));
+
+        Lease lease = asyncResult.get().lease();
+
+        setTimeStatic(clock.get().plus(LockLeaseContract.SERVER_LEASE_TIMEOUT).plus(Duration.ofNanos(1)));
+        assertThat(lease.isValid(leaderClock.time())).isFalse();
+        assertUnlocked(REQUEST_ID);
+    }
+
+    @Test
+    public void refreshShouldReturnAllLockedTokens() {
```

Comment:
we can drop this test, if there's already a test for the case where we didn't refresh all the tests